community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -397,6 +401,10 @@ public Object relationshipGetProperty( long relationshipId, int propertyKeyId )
         {
             return relationship.get().getProperty( propertyKeyId );
         }
+        finally
+        {
+            statement.assertOpen();
+        }
```

Comment:
Because if we read a dynamic property we can not trust the returned value to not be a mix of states caused my record reuse until we check if we are terminated or not.

```
try ( Transaction tx = db.beginTx() )
{
    String value = getSomeNode().getProperty( keyToLongString );
    // Value can not be trusted here
    tx.success();
}
```

This scenario is reproduced by `SlavesCanReadCorruptedData` (now renamed to `TerminationOfSlavesDuringPullUpdatesTest`)